### PR TITLE
PMM-7235 Apply rule filters to the alerts returned by API

### PR DESCRIPTION
**Pull request by artemgavrilov**
[PMM-7235](https://jira.percona.com/browse/PMM-7235)

- [ ] Build: https://github.com/Percona-Lab/pmm-submodules/pull/1447

---
- [x] Tests passed.
- [x] Feature build pass.
- [ ] (Re)requested review.
- [ ] Fix conflicts with target branch.
- [ ] Update API dependency.
- [ ] Update jira ticket description if needed.
- [ ] Attach screenshots/console output to confirm new behavior to jira ticket, if applicable.

### CODE REVIEW

File: services/management/ia/alerts_service.go [hunk 1]
```diff
@@ -129,21 +130,56 @@ func (s *AlertsService) ListAlerts(ctx context.Context, req *iav1beta1.ListAlert
...
+				UpdatedAt: updatedAt,
+			})
+		}
 	}
 
 	return &iav1beta1.ListAlertsResponse{Alerts: res}, nil
 }
 
+// applyFilters applies filters to the given alert, returns true if alert passes all filters.
+func applyFilters(alert *ammodels.GettableAlert, filters []*iav1beta1.Filter) (bool, error) {
```
[the remark this answers is not in the document]
Comment (artemgavrilov) [reply services/management/ia/alerts_service.go:156]: Too java-style as for me)

[the remark this answers is not in the document]
Comment (yareach) [reply services/management/ia/alerts_service.go:156]: That name is not English correct. Try something else.
is ... satisfies, i.e. two verbs.
Can be: is ... satisfied, or does ... satisfy.

[the remark this answers is not in the document]
Comment (artemgavrilov) [reply services/management/ia/alerts_service.go:156]: Does `applyFilters` not clear enough? Do you want me to change it to `isAlertsPassedFilters` or something similar?

[the remark this answers is not in the document]
Comment (artemgavrilov) [reply services/management/ia/alerts_service.go:156]: @oter @yareach  guys, please, I want to move that ticket to QA

**Review by yareach — APPROVED** [review]

File: services/management/ia/alerts_service.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (oter) [reply services/management/ia/alerts_service.go:156]: `satisfiesFilters` would work for me

**Review by oter — APPROVED** [review]

File: services/management/ia/alerts_service.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (artemgavrilov) [reply services/management/ia/alerts_service.go]: Renamed

**Review by BupycHuk — COMMENTED** [review]
Looks good to me, but can we create a test for it?

**Comment by artemgavrilov** [comment]
> Looks good to me, but can we create a test for it?

Yes, we can!

**Review by BupycHuk — APPROVED** [review]

**Pull request merged**